app/Http/Controllers/Admin/FinanceController.php

```diff
@@ -51,7 +51,7 @@ public function index(Request $request)
         # group by the airline
         foreach ($airlines as $airline) {
             # Return all the transactions, grouped by the transaction group
-            $transactions = JournalTransaction::groupBy('transaction_group')
+            $transactions = JournalTransaction::groupBy('transaction_group', 'currency')
```

Comment:
I think this `currency` will break things - what was the exact issue? Do the tests still pass?

Reply:
the page showed a sql exception saying that curerncy field was not in group by clause

Reply:
Can you tell me how to duplicate this? It might be a bug in the package I got this from as well, I'll have to look

Reply:
it happened everytime i clicked on finance in admin panel. Anyway you can't select a field in a query where there are aggregate functions without putting that field in the group by clause too, it's a sql rule....

Reply:
Hmm, I'm not running into it. Can you export your database where it's happening and send it to me? I just want to understand exactly what's going on. Grouping by the `transaction_group` should be enough to limit it. What version of MySQL are you on?

Reply:
Yes, I know that, I've been doing SQL for 10+ years lol. There's already a group and the query should be correct. Adding a second group could make the results actually be incorrect, and the first query you outlined there is what's actually being created. 

So, I'm trying to understand why I'm not seeing an error, but you are. Because if I'm not seeing it but you are, then there's something else that is wrong that's going on, which is what I want to understand, before changing anything. There are a *lot* of small incompatibilities between Maria and MySQL.

Reply:
If you can send me a dump of your database so I can load it and see if I get the error, that would be appreciated. And then I'm doing to install Maria and then Percona to see if I get it on those. This needs to work for everyone, so it's a little more involving on the testing side.

Reply:
What i’m trying to say is that all non aggregate fields must be in the group by clause if there is some aggregate function in the query (and this is a basic and strict rule for all dbs i’ve worked with in the 16 years i’ve made as software developer). 

The fact that MySQL allows that kind of queries is an error on 99% of the other relational dbs around.

Now i’m at work and don’t have my laptop bere so i can’t send tuo the fumo of my db, but i’ll do that tonight when i’m back home.

Anyway you can execute both queries in MySQL and you can see that results are identical.

I find your project very well coded and organized, but, if i can say, the db side of the app is a little bit confusing and difficult to understand and browse (as said before, i’ve worked 16 years as developer, but i’ve many difficulties to understand and find stuff in your db code).

Tonight, or latest tomorrow morning i’ll submit my new work again (you’d only make the assets cause i haven’t found a solution for that on windows). If all is ok than i go straight forward to complete the translation of the app and after that i can look if it’s possibile to reorganize and optimize the db infrastructure, if you want...

Reply:
Yes, I understand how it's supposed to work and I know how SQL works. My point is that we're seeing *different behavior for the same code*. The `groupBy('transaction_group')` is supposed to be generating the correct query, and if it's not then there's a problem either in the ORM layer, or there is difference between Maria and MySQL. I'm trying to understand why exactly the original code isn't working *before* changing anything. Without understanding why it's broken or doesn't work can cause issues later. Remember, this needs to work for *everyone*. 

I have never had to put any non-aggregate fields into the `GROUP BY` and that would cause different behavior. For example, in this case, the currency is set to `USD`, with a transaction group of `Gate Costs`, for example. If you add a second group-by, if there are transactions of a different currency, they will be grouped separately, which is not what we want in this case.

If you do an `EXPLAIN`, the underlying query will be different. I haven't tried it yet. But I want to understand why we're seeing different behavior. Does that make sense?

We can talk on Discord about the DB stuff. It's using a standard Repository-pattern on top of the ORM/models layer. Most queries go through the repository layer first, in order to take advantage of caching, without having to wrap individual calls to the models in caching. The repository also enables some rudimentary searching and filtering, and also allows caching these.

Reply:
And what's your value for this:

```sql
SELECT @@SQL_MODE, @@GLOBAL.SQL_MODE;
```

Reply:
I'm just trying to see why exactly MySQL is treating this differently and it's version specific. It's basically for my own edification. 

If you can fix the PR so the tests are also passing, I'll merge it all in.

Reply:
@@SQL_MODE = STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION

@@GLOBAL.SQL_MODE = STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION

Here is my dump:

[Dump12052018.zip](https://github.com/nabeelio/phpvms/files/2014820/Dump12052018.zip)

So seems to me that the error is somewhere in the app's db layer or in the MySQL PDO cause on mariaDB it generates this query:

SELECT transaction_group, currency, [...] GROUP BY transaction_group

and it's obviously wrong....

Reply:
Thanks. Yeah I want to just understand if it's a problem in the ORM or in PDO. Because if there's a bug here then it'll be doing the same in other calls. Thanks!